homeassistant/components/overkiz/alarm_control_panel.py

```diff
@@ -0,0 +1,304 @@
+        ),
+    ),
+    # MyFoxAlarmController
+    OverkizAlarmDescription(
+        key=UIWidget.MY_FOX_ALARM_CONTROLLER,
+        supported_features=SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_NIGHT,
+        fn_state=_state_myfox_alarm_controller,
+        alarm_disarm=lambda execute_command: execute_command(OverkizCommand.DISARM),
+        alarm_arm_night=lambda execute_command: execute_command(OverkizCommand.PARTIAL),
+        alarm_arm_away=lambda execute_command: execute_command(OverkizCommand.ARM),
+    ),
+    # AlarmPanelController
+    OverkizAlarmDescription(
+        key=UIWidget.ALARM_PANEL_CONTROLLER,
+        supported_features=(
+            SUPPORT_ALARM_ARM_AWAY | SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_NIGHT
+        ),
+        fn_state=_state_alarm_panel_controller,
+        alarm_disarm=lambda execute_command: execute_command(OverkizCommand.DISARM),
+        alarm_arm_home=lambda execute_command: execute_command(
+            OverkizCommand.ARM_PARTIAL_DAY
+        ),
+        alarm_arm_night=lambda execute_command: execute_command(
+            OverkizCommand.ARM_PARTIAL_NIGHT
+        ),
+        alarm_arm_away=lambda execute_command: execute_command(OverkizCommand.ARM),
+    ),
+]
+
+SUPPORTED_DEVICES = {description.key: description for description in ALARM_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz alarm control panel from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizAlarmControlPanel] = []
+
+    for device in data.platforms[Platform.ALARM_CONTROL_PANEL]:
+        if description := SUPPORTED_DEVICES.get(device.widget) or SUPPORTED_DEVICES.get(
+            device.ui_class
+        ):
+            entities.append(
+                OverkizAlarmControlPanel(
+                    device.device_url,
+                    data.coordinator,
+                    description,
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class OverkizAlarmControlPanel(OverkizDescriptiveEntity, AlarmControlPanelEntity):
+    """Representation of an Overkiz Alarm Control Panel."""
+
+    entity_description: OverkizAlarmDescription
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self.entity_description.supported_features
+
+    @property
+    def state(self) -> str:
+        """Return the state of the device."""
+        return self.entity_description.fn_state(self.executor.select_state)
+
+    async def async_alarm_disarm(self, code: str | None = None) -> None:
+        """Send disarm command."""
+        if self.entity_description.alarm_disarm:
+            await self.entity_description.alarm_disarm(
+                self.executor.async_execute_command
+            )
```

Comment:
And in the base entity (so it can be shared with `select` and `switch`)
```suggestion
        await self.async_execute_command(
            self.entity_description.alarm_disarm,
            self.entity_description.alarm_disarm_args,
        )
```

And in the base entity, you define a method that can be shared with `select` and `switch`:
```python
    async def async_execute_command(self, command_name: str | None, args: Any) -> None:
        """Check command name and send to the executor."""
        if command_name is None:
            return
        if args is UNDEFINED:
            await self.executor.async_execute_command(command_name)
        await self.executor.async_execute_command(command_name, *args)
```

Reply:
Didn't add it to the base entity yet, since this refactoring hasn't been done yet for switch.